bin/anthology/utils.py

```diff
@@ -59,18 +59,30 @@ def build_anthology_id(collection_id, volume_id, paper_id):
 
 def deconstruct_anthology_id(anthology_id):
     """
-    Transforms an Anthology id into its constituent collection id, voulume id, and paper id
+    Transforms an Anthology id into its constituent collection id, volume id, and paper id
     parts. e.g,
 
         P18-1007 -> ('P18', '1', '7')
         W18-6310 -> ('W18', '63', '10')
+
+    Also can deconstruct Anthology volumes 
+
+        P18-1 -> ('P18', '1', nil)
+        W18-63 -> ('W18', '63', nil)
     """
+
     collection_id, rest = anthology_id.split('-')
+    assert len(collection_id) == 3, "Collection IDs should be 1 letter prefix + 2 digit year"
     if collection_id.startswith('W') or collection_id.startswith('C69'):
-        return (collection_id, str(int(rest[0:2])), str(int(rest[2:])))
+        try:
+            return (collection_id, str(int(rest[0:2])), str(int(rest[2:])))
+        except ValueError: # Possible Volume only identifier
+            return (collection_id, str(int(rest[0:2])), None)
     else:
-        return (collection_id, str(int(rest[0:1])), str(int(rest[1:])))
-
+        try:
+            return (collection_id, str(int(rest[0:1])), str(int(rest[1:])))
+        except ValueError: # Possible Volume only identifier
+            return (collection_id, str(int(rest[0:2])), None)
```

Comment:
Typo here? Should be `str(int(rest[:1]))`, no?

Reply:
Yep.  Thanks for the spot.  Reworked without Exception handling in 80073b4.

Reply:
The reworked logic looks better to me in this context, thanks!